src/worker.js

```diff
@@ -194,13 +210,9 @@ self.onmessage = (e) => {
       }
 #if MODULARIZE
 #if MINIMAL_RUNTIME
-      {{{ EXPORT_NAME }}}(imports).then(function (instance) {
-        Module = instance;
-      });
+      {{{ EXPORT_NAME }}}(imports);
 #else
-      {{{ EXPORT_NAME }}}(Module).then(function (instance) {
-        Module = instance;
-      });
+      {{{ EXPORT_NAME }}}(Module);
```

Comment:
I wonder if you could factor out the startWorker helper that just sets `Module = instance` and then sends `loaded`?

That seems simpler NFC that might make this change simpler.

Reply:
I actually tried that earlier today in https://github.com/emscripten-core/emscripten/commit/3935cdcc57409e05830be9603e8eb0b906809f40 as a potential fix for flakiness, if that's what you mean.

It didn't pan as a fix for flaky FF tests, but I suppose I could submit it separately anyway?

Reply:
Given that this change is +90/-64, and some of it is indentation, I'm not sure if it's worth the time splitting / reviewing them separately though.

Reply:
I just always like smaller PRs, but in this case its not a big deal and the two changes are related, so we can leave it as one if you prefer.

Reply:
I don't care too much either way.

Reply:
Extracted in https://github.com/emscripten-core/emscripten/pull/18313.